### ISSUE TITLE
[Docs] Sidebar toggle issue

### ISSUE THREAD

**Issue by ciajulius** [Semantic-Org/Semantic-UI#5317]
The issue you can see [in this template](https://semantic-ui.com/examples/homepage.html), when viewing in mobile pressing the three bars (sidebar icon) to open sidebar, and then again pressing those same three bars (sidebar icon) to close sidebar, it does not close. It closes after you press again those three bars (sidebar icon). I did reproduce this issue with chrome and my android phone.
In template, 2.2.4 version is used, so I switched to 2.2.10, but same result.
I set debug to true, and this is what happens when you try to close sidebar, but it does not close
![semantic_ui_sidebar_toggle_issue](https://cloud.githubusercontent.com/assets/7648634/25434096/a2eec7f0-2a93-11e7-9e48-f264b886c07c.png)

**Comment by Banandrew** [comment]
Hi @ciajulius, I didn’t quite understand the issue, it closes only the second time you press the sidebar icon, but not the first? Did you reproduce it on the desktop version of Chrome, or on Chrome for Android? What version of Chrome/Android were you using?

**Comment by ciajulius** [comment]
I'm using android 6.0.1, and chrome version is 57.0.2987.132
When in that template I tap on sidebar icon to open sidebar, it opens, when for the second time I tap sidebar icon to close it, it does not close (in previous screenshot you can see what happened on that tap when it did not close (it tried to close and immediately opened up again)). When I tap again on sidebar icon - sidebar closes. If I would tap somewhere else, not on sidebar icon, sidebar would close as expected (from the first time).

**Comment by ciajulius** [comment]
To reproduce:
1. Open chrome browser on your computer
2. [Go here](https://semantic-ui.com/examples/homepage.html)
3. Open Developer tools
4. Click "toggle device toolbar"
5. Tap on sidebar icon (this will open sidebar)
6. Tap on sidebar icon (in my case, this does not close sidebar)

**Comment by Banandrew** [comment]
@ciajulius got it, thank you.

For the future reference, if you can’t see it, repeat it a couple of times.

**Comment by ciajulius** [comment]
@Banandrew if this is not a bug, what was done incorrectly that lead to this issue?

**Comment by Banandrew** [comment]
@ciajulius I don’t know, I just thought that I labeled it prematurely. I use a VM for development, and when I ran the example in my local environment, I couldn’t reproduce the issue: http://i.imgur.com/J8eGWVP.gifv I can do this indefinitely, and it won’t show up, but I do see it on SUI’s website—it closes and opens the sidebar almost immediately every fifth click or so.

Have you tried replicating the example locally?

**Comment by ciajulius** [comment]
@Banandrew no, I didn't. I use minified files from cdn, since I'm kinda new to semantic ui, I used the same template as the one with the issue, that's how I found this issue today. Tried a few different versions of semantic, did not help.

I just checked SUI's doc's with my android phone, sidebar examples page, in one of examples I can reproduce this issue. Example that I'm talking about, has title "Triggering show/hide with other content", I pressed "run code", and tapping on "trigger sidebar" button makes sidebar misbehave the same way, but as you noticed before, this happens not every time the button is being pressed

**Comment by Banandrew** [comment]
@ciajulius I sat down to figure it out, so what happens is the button that triggers the sidebar is clickable when the sidebar is being shown, and when you click anywhere outside of the sidebar it closes, but since the button is also clickable, the sidebar opens immediately after being closed.

This behaviour is intended, and it’s a developer’s responsibility to address clickable elements. If you’ll have questions in the future, you can always ask them in our [Gitter chatroom](https://gitter.im/Semantic-Org/Semantic-UI).

**Comment by ciajulius** [comment]
@Banandrew Well, I still think this is a bug. Why doesn't this happen every time then? Why doesn't this happen when I'm using laptop without chrome developer tools?
Attach events defaults to toggle, so it should open when given element is pressed, and close when given element is pressed again. One action at a time, but not both.

**Comment by Banandrew** [comment]
@ciajulius You’re right, sorry about that.

**Comment by lvbeck** [comment]
I have same issue, the sidebar('hide') behavior is not working on iOS devices

**Comment by Clindbergh** [comment]
I had the same issue on a page of mine mobile devices. It worked when I explicitly called `sidebar('hide')` e.g.
```
$sidebar.find('a.item').click(function(){
       $sidebar.sidebar('hide');
});
```
#3285 #1683 should be marked as duplicates.

**Comment by SarhadSalam** [comment]
I faced the exact same issue, for me it happened on all mobile devices and tablets, however to fix it, i used the callback functions to set up a local boolean var in my  js. I am using the latest version of semantic-ui.

The `open` is the boolean variable that indicated the status of the sidebar, and this worked perfectly! (sidebar is just the function i had since i was using it too many times). 

  `open ? sidebar('show') : sidebar('hide');`


Although, I do think this is a bug and beats the purpose of having a `toggle` if hacks like this are needed.

**Comment by FriendOfTheNight** [comment]
Any update on this? I would call this a bug as well. If default behavior is toggle then it should toggle!

**Comment by SWAT10101** [comment]
I have same issue in version 2.4.1
